src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.FormatInfo.cs

```diff
@@ -33,45 +33,50 @@ internal struct MountPointInformation
         [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetSpaceInfoForMountPoint", SetLastError = true)]
         internal static partial int GetSpaceInfoForMountPoint([MarshalAs(UnmanagedType.LPUTF8Str)] string name, out MountPointInformation mpi);
 
-        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetFormatInfoForMountPoint", SetLastError = true)]
-        internal static unsafe partial int GetFormatInfoForMountPoint(
-            [MarshalAs(UnmanagedType.LPUTF8Str)] string name,
-            byte* formatNameBuffer,
-            int bufferLength,
-            long* formatType);
-
-        internal static int GetFormatInfoForMountPoint(string name, out string format)
-        {
-            return GetFormatInfoForMountPoint(name, out format, out _);
-        }
-
-        internal static int GetFormatInfoForMountPoint(string name, out DriveType type)
+        internal static unsafe Error GetFileSystemTypeNameForMountPoint(string name, out string format)
         {
-            return GetFormatInfoForMountPoint(name, out _, out type);
-        }
-
-        private static unsafe int GetFormatInfoForMountPoint(string name, out string format, out DriveType type)
-        {
-            byte* formatBuffer = stackalloc byte[MountPointFormatBufferSizeInBytes];    // format names should be small
-            long numericFormat;
-            int result = GetFormatInfoForMountPoint(name, formatBuffer, MountPointFormatBufferSizeInBytes, &numericFormat);
-            if (result == 0)
+            if (OperatingSystem.IsLinux())
             {
-                // Check if we have a numeric answer or string
-                format = numericFormat != -1 ?
-                    Enum.GetName(typeof(UnixFileSystemTypes), numericFormat) ?? string.Empty :
-                    Marshal.PtrToStringUTF8((IntPtr)formatBuffer)!;
-                type = GetDriveType(format);
+                // Canonicalize and resolve symbolic links.
+                string? path = Sys.RealPath(name);
```

Comment:
This function is now going to make several syscalls where there used to be one.

If this turns out to be an performance issue, we can reduce the nr of calls:

- If we use `statx` then we can get the `STATX_MNT_ID` which corresponds to the first field in `mountinfo`. This replaces multiple syscalls made by `realpath` into a single syscall.

- For Linux 6.8+, we can call `statx` with `STATX_MNT_ID_UNIQUE` and then use that with the `statmount` syscall with `STATMOUNT_FS_TYPE ` which avoids the syscalls for opening and reading `mountinfo` to get the filesystem type name.

I don't intend to include these changes as part of this PR (unless someone finds it important).